vaas-app/src/vaas/validators.py

```diff
@@ -2,7 +2,26 @@
 
 import re
 
-from django.core.validators import RegexValidator
+from django.core.validators import RegexValidator, ValidationError
+from django.conf import settings
 
 vcl_name_re = re.compile(r'^[a-zA-Z0-9_]+$')
 vcl_name_validator = RegexValidator(vcl_name_re, "Allowed characters: letters, numbers and underscores.", 'invalid')
+
+
+class VclVariableValidator(ValidationError):
+    pass
+
+
+def vcl_variable_validator(vcl_content, vcl_pk, vcl_variable_class, varnish_server_class):
+
+    pattern = re.compile('#{.*}', re.MULTILINE)
+    placeholders = re.findall(pattern, vcl_content)
+    placeholders = [p[2:-1] for p in placeholders]
```

Comment:
nonoptimal data structure - placeholders can be repeated in list - you can use set

>>> pattern = re.compile('a')
>>> string = 'abceabcdabcd'
>>> re.findall(pattern, string)
['a', 'a', 'a']